src/sys/mman.rs

```diff
@@ -481,33 +480,33 @@ pub unsafe fn mmap_anonymous(
 /// detailed requirements.
 #[cfg(any(target_os = "linux", target_os = "netbsd"))]
 pub unsafe fn mremap(
-    addr: *mut c_void,
+    addr: NonNull<c_void>,
     old_size: size_t,
     new_size: size_t,
     flags: MRemapFlags,
-    new_address: Option<*mut c_void>,
-) -> Result<*mut c_void> {
+    new_address: Option<NonNull<c_void>>,
+) -> Result<NonNull<c_void>> {
     #[cfg(target_os = "linux")]
     let ret = libc::mremap(
-        addr,
+        addr.as_ptr(),
         old_size,
         new_size,
         flags.bits(),
-        new_address.unwrap_or(std::ptr::null_mut()),
+        new_address.map(NonNull::as_ptr).unwrap_or(std::ptr::null_mut()),
     );
     #[cfg(target_os = "netbsd")]
     let ret = libc::mremap(
-        addr,
+        addr.as_ptr(),
         old_size,
-        new_address.unwrap_or(std::ptr::null_mut()),
+        new_address.map(NonNull::as_ptr).unwrap_or(std::ptr::null_mut()),
         new_size,
         flags.bits(),
     );
 
     if ret == libc::MAP_FAILED {
         Err(Errno::last())
     } else {
-        Ok(ret)
+        Ok(unsafe { NonNull::new_unchecked(ret) })
```

Comment:
And this one: `mremap()` will never return a NULL pointer, it is safe to use `NonNull::new_unchecked()` here

Reply:
Added comment.